<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="windowTitle">
   <string>SpeckMobil</string>
  </property>
  <property name="windowIcon">
   <iconset resource="icons.qrc">
    <normaloff>:/resources/icon.png</normaloff>:/resources/icon.png</iconset>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout_5">
    <item row="0" column="0">
     <widget class="QPushButton" name="pushButton_connect">
      <property name="text">
       <string>Connect to ELM</string>
      </property>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QPushButton" name="pushButton_findModules">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="text">
       <string>Find Modules</string>
      </property>
     </widget>
    </item>
    <item row="0" column="2" rowspan="3">
     <widget class="QGroupBox" name="groupBox">
      <property name="title">
       <string>DTC Handling</string>
      </property>
      <layout class="QGridLayout" name="gridLayout">
       <property name="leftMargin">
        <number>2</number>
       </property>
       <property name="topMargin">
        <number>2</number>
       </property>
       <property name="rightMargin">
        <number>2</number>
       </property>
       <property name="bottomMargin">
        <number>2</number>
       </property>
       <item row="0" column="0">
        <layout class="QGridLayout" name="gridLayout_3">
         <item row="1" column="2">
          <layout class="QHBoxLayout" name="horizontalLayout_2">
           <item>
            <widget class="QLabel" name="label_9">
             <property name="text">
              <string>Number of Errors:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="lineEdit_numDTC">
             <property name="enabled">
              <bool>true</bool>
             </property>
             <property name="sizePolicy">
              <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="maximumSize">
              <size>
               <width>30</width>
               <height>16777215</height>
              </size>
             </property>
             <property name="inputMask">
              <string>hH</string>
             </property>
             <property name="maxLength">
              <number>2</number>
             </property>
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item row="0" column="0" colspan="2">
          <widget class="QPushButton" name="pushButton_readErrors">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>Read Errors in Module</string>
           </property>
          </widget>
         </item>
         <item row="0" column="2">
          <widget class="QPushButton" name="pushButton_resetErrors">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>Clear Errors in Module</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0" colspan="2">
          <widget class="QPushButton" name="pushButton_resetErrors_2">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>Clear Errors in all Modules</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_10">
           <property name="text">
            <string>Error Codes:</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1" colspan="2">
          <widget class="QLineEdit" name="lineEdit_DTCs"/>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QPushButton" name="pushButton_openModule">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="sizePolicy">
       <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="toolTip">
       <string>Opens a connection to the specfied module</string>
      </property>
      <property name="text">
       <string>Open Module</string>
      </property>
      <property name="icon">
       <iconset resource="icons.qrc">
        <normaloff>:/resources/red.png</normaloff>:/resources/red.png</iconset>
      </property>
      <property name="iconSize">
       <size>
        <width>12</width>
        <height>12</height>
       </size>
      </property>
     </widget>
    </item>
    <item row="1" column="1">
     <widget class="QPushButton" name="pushButton_startDiag">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="text">
       <string>Read Info</string>
      </property>
     </widget>
    </item>
    <item row="2" column="0" rowspan="2" colspan="2">
     <widget class="QGroupBox" name="groupBox_2">
      <property name="title">
       <string>Choose Module</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_2">
       <property name="leftMargin">
        <number>2</number>
       </property>
       <property name="topMargin">
        <number>2</number>
       </property>
       <property name="rightMargin">
        <number>2</number>
       </property>
       <property name="bottomMargin">
        <number>2</number>
       </property>
       <item row="0" column="0">
        <layout class="QGridLayout" name="gridLayout_4">
         <item row="0" column="0">
          <widget class="QComboBox" name="comboBox_modules">
           <property name="sizePolicy">
            <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="toolTip">
            <string>Select a module to connect to</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <layout class="QHBoxLayout" name="horizontalLayout">
           <item>
            <widget class="QLabel" name="label_2">
             <property name="text">
              <string>Module address:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="lineEdit_moduleAddress">
             <property name="enabled">
              <bool>true</bool>
             </property>
             <property name="sizePolicy">
              <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="maximumSize">
              <size>
               <width>30</width>
               <height>16777215</height>
              </size>
             </property>
             <property name="inputMask">
              <string>hH</string>
             </property>
             <property name="text">
              <string>01</string>
             </property>
             <property name="maxLength">
              <number>2</number>
             </property>
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item row="2" column="0">
          <layout class="QHBoxLayout" name="horizontalLayout_13">
           <item>
            <widget class="QLabel" name="label_3">
             <property name="text">
              <string>Module number:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="lineEdit_moduleNum">
             <property name="enabled">
              <bool>true</bool>
             </property>
             <property name="sizePolicy">
              <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="maximumSize">
              <size>
               <width>30</width>
               <height>16777215</height>
              </size>
             </property>
             <property name="inputMask">
              <string>hH</string>
             </property>
             <property name="text">
              <string>01</string>
             </property>
             <property name="maxLength">
              <number>2</number>
             </property>
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
             <property name="readOnly">
              <bool>true</bool>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item row="3" column="2" rowspan="2">
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QLabel" name="label_8">
        <property name="text">
         <string>Shop Number:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="lineEdit_shopNum"/>
      </item>
      <item>
       <widget class="QLabel" name="label_6">
        <property name="text">
         <string>Coding:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="lineEdit_coding"/>
      </item>
     </layout>
    </item>
    <item row="4" column="0" rowspan="2" colspan="2">
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QLabel" name="label_13">
        <property name="text">
         <string>Module identification:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="lineEdit_moduleId">
        <property name="toolTip">
         <string>Displays information about the current module</string>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_14">
        <property name="text">
         <string>Hardware part number:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="lineEdit_hwNum">
        <property name="toolTip">
         <string>Displays information about the current Electronic Control Module</string>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Software part number:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="lineEdit_swNum"/>
      </item>
      <item>
       <widget class="QLabel" name="label_7">
        <property name="text">
         <string>Software Version:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="lineEdit_swVers"/>
      </item>
      <item>
       <widget class="QLabel" name="label_11">
        <property name="text">
         <string>Serial number:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="lineEdit_serNum"/>
      </item>
      <item>
       <widget class="QLabel" name="label_12">
        <property name="text">
         <string>VIN:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="lineEdit_vin"/>
      </item>
     </layout>
    </item>
    <item row="5" column="2">
     <widget class="QPlainTextEdit" name="plainTextEdit_log"/>
    </item>
   </layout>
   <zorder>label</zorder>
   <zorder>label_13</zorder>
   <zorder>lineEdit_moduleId</zorder>
   <zorder>lineEdit_hwNum</zorder>
   <zorder>lineEdit_coding</zorder>
   <zorder>label_14</zorder>
   <zorder>lineEdit_swNum</zorder>
   <zorder>label_6</zorder>
   <zorder>pushButton_readErrors</zorder>
   <zorder>plainTextEdit_log</zorder>
   <zorder>comboBox_modules</zorder>
   <zorder>label_7</zorder>
   <zorder>lineEdit_swVers</zorder>
   <zorder>label_8</zorder>
   <zorder>lineEdit_shopNum</zorder>
   <zorder>pushButton_resetErrors_2</zorder>
   <zorder>lineEdit_numDTC</zorder>
   <zorder>label_9</zorder>
   <zorder>lineEdit_DTCs</zorder>
   <zorder>label_10</zorder>
   <zorder>label_11</zorder>
   <zorder>lineEdit_serNum</zorder>
   <zorder>label_12</zorder>
   <zorder>lineEdit_vin</zorder>
   <zorder>label_9</zorder>
   <zorder>lineEdit_numDTC</zorder>
   <zorder>groupBox</zorder>
   <zorder>groupBox_2</zorder>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>497</width>
     <height>18</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_File">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menu_Options">
    <property name="title">
     <string>&amp;Options</string>
    </property>
    <addaction name="actionApplication_settings"/>
    <addaction name="actionSerial_port_settings"/>
    <addaction name="separator"/>
    <addaction name="actionClear_log"/>
   </widget>
   <widget class="QMenu" name="menu_Help">
    <property name="title">
     <string>&amp;Help</string>
    </property>
    <addaction name="action_About"/>
   </widget>
   <addaction name="menu_File"/>
   <addaction name="menu_Options"/>
   <addaction name="menu_Help"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionExit">
   <property name="text">
    <string>E&amp;xit</string>
   </property>
  </action>
  <action name="actionSerial_port_settings">
   <property name="text">
    <string>&amp;Serial port settings</string>
   </property>
  </action>
  <action name="action_About">
   <property name="text">
    <string>&amp;About</string>
   </property>
  </action>
  <action name="actionClear_log">
   <property name="text">
    <string>&amp;Clear log</string>
   </property>
  </action>
  <action name="actionApplication_settings">
   <property name="text">
    <string>&amp;Application settings</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="icons.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionExit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>453</x>
     <y>358</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
